### ISSUE TITLE
Glowroot randomly drops anonymous account

### ISSUE THREAD

**Issue by oboukili** [glowroot/glowroot#613]
Hello,

After some time (days or weeks) our glowroot UI prompts for a login/password even though we did not set up any account on startup (through the setup-admin-user flag).
  
The workaround posted here https://github.com/glowroot/glowroot/issues/428#issuecomment-406336350 (dropping user and role tables, and restarting the central collector) is working, but what puzzles us is that those tables are already empty.

Any advices?
Thanks!

EDIT: using 0.13.4, we will deploy 0.13.5 this afternoon.

**Comment by trask** [comment]
Hi @oboukili, during startup, if the central collector sees that the user and role tables are empty, it will create an `anonymous` user and `Administrator` role. It sounds like somehow those records are getting deleted at some point after startup?

**Comment by oboukili** [comment]
hi, yes exactly!

**Comment by trask** [comment]
I'm not sure how that could be happening, are you running a Cassandra cluster? maybe it's a consistency issue?

**Comment by oboukili** [comment]
Yes we are running a 3 nodes cluster. Now that you've mentioned it, I don't know why I haven't thought about that before. I'll report back on that ticket later, thanks!

**Comment by jeromebridge** [comment]
Any further updates on this issue. I'm experiencing a similar issue running a Cassandra cluster on Kubernetes.

**Comment by grib0** [comment]
Same here. I drop cassandra database and try to restart normally to have database recreated by glowroot. Then I restart with `setup-admin-user <username> <password>` like explained https://github.com/glowroot/glowroot/issues/428#issuecomment-406336350 but got

```
2023-06-02 15:16:14,216 INFO  [org.glowroot] connected to Cassandra (version 3.11.14), using keyspace 'glowrootdev' (replication factor 1) and consistency level QUORUM
2023-06-02 15:16:14,964 INFO  [org.glowroot] running setup-admin-user
2023-06-02 15:16:14,967 ERROR [org.glowroot] Administrator role does not exist, exiting
```

**Comment by Never-lab** [comment]
Anonymous/shared UI account “randomly dropped” — likely session/auth/cookie or multi-Central sticky issue. Needs modern repro on 0.14.x (embedded vs Central).

@nowheresly — needs-info or close stale.
